x-pack/metricbeat/module/aws/awshealth/awshealth.go

```diff
@@ -0,0 +1,431 @@
+			"event_description":          hd.eventDescription,
+		},
+		RootFields: mapstr.M{
+			"cloud.provider": "aws",
+		},
+		ID: generateEventID(eventID),
+	}
+	return event
+}
+
+type HealthDetails struct {
+	event                  types.Event
+	eventDescription       string
+	affectedEntities       []types.AffectedEntity
+	affectedEntityPending  int32
+	affectedEntityResolved int32
+	affectedEntityOthers   int32
+}
+
+type AffectedEntityDetails struct {
+	AwsAccountId    string    `json:"aws_account_id"`
+	EntityUrl       string    `json:"entity_url"`
+	EntityValue     string    `json:"entity_value"`
+	LastUpdatedTime time.Time `json:"last_updated_time"`
+	StatusCode      string    `json:"status_code"`
+	EntityArn       string    `json:"entity_arn"`
+}
+
+// getValueOrDefault returns the dereferenced value of a pointer v of any type T.
+// If the pointer is nil, it returns the specified defaultValue of type T.
+func getValueOrDefault[T any](v *T, defaultValue T) T {
+	if v != nil {
+		return *v
+	}
+	return defaultValue
+}
+
+func createAffectedEntityDetails(affectedEntities []types.AffectedEntity) []AffectedEntityDetails {
+	aed := make([]AffectedEntityDetails, len(affectedEntities))
+	for i := range affectedEntities {
+		aed[i] = AffectedEntityDetails{
+			AwsAccountId:    getValueOrDefault(affectedEntities[i].AwsAccountId, ""),
+			EntityUrl:       getValueOrDefault(affectedEntities[i].EntityUrl, ""),
+			EntityValue:     getValueOrDefault(affectedEntities[i].EntityValue, ""),
+			LastUpdatedTime: getValueOrDefault(affectedEntities[i].LastUpdatedTime, time.Time{}),
+			StatusCode:      getValueOrDefault((*string)(&affectedEntities[i].StatusCode), ""),
+			EntityArn:       getValueOrDefault(affectedEntities[i].EntityArn, ""),
+		}
+	}
+	return aed
+}
+
+// generateEventID hashes the provided eventID and returns the first 20 characters.
+// This is used to generate a unique but consistent event ID prefix.
+func generateEventID(eventID string) string {
+	h := sha256.New()
+	h.Write([]byte(eventID))
+	prefix := hex.EncodeToString(h.Sum(nil))
+	return prefix[:20]
+}
+
+// getEventsSummary retrieves a summary of AWS Health events that meet the specified
+// filter criteria. It uses the DescribeEvents API to get a list of events. For each
+// event, it calls getDescribeEventDetails with the EventARN to get details like details
+// of affected entities by calling DescribeAffectedEntities API. The DescribeAffectedEntities
+// is called to fetch the description of the event.
+// The function returns a slice of mb.Event structs containing the summarized event info.
+func (m *MetricSet) getDescribeEventDetails(ctx context.Context, awsHealth *health.Client, event types.Event, ch chan<- HealthDetails) error {
+	hd := HealthDetails{event: event, affectedEntities: []types.AffectedEntity{}}
+	eventDetails, err := awsHealth.DescribeEventDetails(ctx, &health.DescribeEventDetailsInput{
+		EventArns: []string{*event.Arn},
```

Comment:
what's the reason we are doing this one event at a time (same question for `DescribeAffectedEntities` below)?